test/util/fake-timers-test.js

```diff
@@ -1066,11 +1066,79 @@ describe("fakeTimers.clock", function () {
         });
 
         it("does not fake methods not provided", function () {
-            this.clock = fakeTimers.useFakeTimers("setTimeout", "Date", "setImmediate");
+            this.clock = fakeTimers.useFakeTimers({toFake: ["setTimeout", "Date", "setImmediate"]});
 
             assert.same(clearTimeout, fakeTimers.timers.clearTimeout);
             assert.same(setInterval, fakeTimers.timers.setInterval);
             assert.same(clearInterval, fakeTimers.timers.clearInterval);
         });
+
+        it("installs by default without nextTick", function () {
+            this.clock = fakeTimers.useFakeTimers();
+            var called = false;
+            process.nextTick(function () {
+                called = true;
+            });
+            this.clock.runAll();
+            assert(!called);
+            this.clock.restore();
+        });
+
+        it("installs with nextTick", function () {
+            this.clock = fakeTimers.useFakeTimers({toFake: ["nextTick"]});
+            var called = false;
+            process.nextTick(function () {
+                called = true;
+            });
+            this.clock.runAll();
+            assert(called);
+            this.clock.restore();
+        });
+
+        it("installs clock in advancing mode and triggers setTimeout", function (done) {
+            this.clock = fakeTimers.useFakeTimers({shouldAdvanceTime: true});
+            this.clock.setTimeout(function () {
+                this.clock.restore();
+                done();
+            }.bind(this), 10);
+        });
+
+        it("installs clock in advancing mode and triggers setInterval", function (done) {
+            this.clock = fakeTimers.useFakeTimers({shouldAdvanceTime: true});
+            var counter = 0;
+            var iterations = 3;
+            var id = this.clock.setInterval(function () {
+                if (counter++ < iterations) {return;}
+                this.clock.clearInterval(id);
+                this.clock.restore();
+                done();
+            }.bind(this), 10);
+        });
+
+        it("installs clock in advancing mode and triggers setImmediate", function (done) {
+            this.clock = fakeTimers.useFakeTimers({shouldAdvanceTime: true});
+            this.clock.setImmediate(function () {
+                this.clock.restore();
+                done();
+            }.bind(this));
+        });
+
+        it("throws on old useFakeTimers signatures", function () {
+            assert.exception(function () {
+                fakeTimers.useFakeTimers("setImmediate");
+            }, { name: "TypeError" });
```

Comment:
[`assert.exception`](http://docs.busterjs.org/en/latest/modules/referee/#exception) also supports matching on the message, which would make for sturdier tests